### PR TITLE
dev/core#3920 Improve footer and support menu

### PR DESCRIPTION
**Pull request by larssandergreen**
Overview
----------------------------------------
I believe this incorporates all the feedback from the issue and from chat to simplify the footer. Since this removes the issues and documentation links from the footer, I've also given the Support menu a once over and updated it in line with what is found on [civicrm.org](http://civicrm.org/) and in the Resources dashlet.

I think it's nice to keep the footer to one line and incorporate all the info (CiviCRM link, version, GNU notice and link) in one sentence. Consensus in the issue seemed to be that it is OK to remove the links from the second line as they are covered in the Support Menu (and Download is not needed).

A few questions:

~~1. Should the link for the User Guide be redirected through civicrm.org, like the Documentation and Developer Docs links are?~~ (@homotechsual says no on chat)
2. Should the link for Stack Exchange also be redirected through civicrm.org, just in case it changes in the future?
3. What is the general strategy for updating the menu for existing installations? It would be nice to give all sites these more complete support links, but that will require adding, perhaps changing the labels, and possibly re-ordering if we want them to end up in any kind of logical order. Probably very few have changed the Support menu, but of course I don't want to mess anything up. I could just add the new options at the bottom, which is quite safe, but also creates a kind of weird menu ordering.
I notice on our install we still have a link to the archived forums and two different links to documentation in there, so updates would be nice! Happy to take a crack at some SQL for upgrades, but would like some guidance on general approach here first.

Before
----------------------------------------
Footer is busy and duplicates the support menu.
System Status is only present sometimes and styling in inconsistent depending on status.
image

Support menu is incomplete.
image

After
----------------------------------------
Footer is simpler and does not duplicate support menu.

When System Status is not available due to the vagaries of caching, if you have administer CiviCRM, you'll just see a System Status link, so at least we're consistently showing at least a link here (this one is for @elisseck). The System Status span is now consistently styled, no matter what status it is.
image
With links:
[System Status](http://drupal-test.localhost/civicrm/a/#/status)   Powered by [CiviCRM](https://civicrm.org/) 5.57.alpha1, free and open source [AGPLv3](http://www.gnu.org/licenses/agpl-3.0.html) software.

For the Support menu, I added additional items to cover all options that are available.
I think a direct link to the User Guide is a good idea, since that should be the first stop for users — so let's make it prominent and easy to get to. Get started is not needed as it is about things you'd do before installing CiviCRM, which is not relevant while you are already using CiviCRM.

### CODE REVIEW

File: templates/CRM/common/footer.tpl [hunk 1]
```diff
@@ -14,18 +14,17 @@
...
-    {ts}Powered by CiviCRM{/ts} <a href="https://download.civicrm.org/about/{$version}" rel="external" target="_blank">{$version}</a>.
     {if $footer_status_severity}
-      <span class="status{if $footer_status_severity gt 3} crm-error{elseif $footer_status_severity gt 2} crm-warning{else} crm-ok{/if}">
+    <span class="status{if $footer_status_severity gt 3} crm-error{elseif $footer_status_severity gt 2} crm-warning{else} crm-ok{/if}">
       <a href="{crmURL p='civicrm/a/#/status'}">{$footer_status_message}</a>
-    </span>
+    </span>&nbsp;
+    {elseif call_user_func(array('CRM_Core_Permission','check'), 'administer CiviCRM')}
+    <span class="status crm-status-none">
+      <a href="{crmURL p='civicrm/a/#/status'}">System Status</a>
```
Comment (mlutfy) [inline templates/CRM/common/footer.tpl:23]: please wrap the string in `{ts}System Status{/ts}` for translation

**Comment by JoeMurray** [comment]
Thanks for your contribution here, @larssandergreen !

**Comment by mlutfy** [comment]
+1 these small details are important.

Pure bike shed, feel free  to ignore: personally I would place the System Status at the end (not at the beginning), because it looks a bit odd (but maybe that's the point).

For redirects:

- docs: I agree with Mikey
- Stack Exchange: personally I would prefer a way to have some stats on how many people click that link. We don't really collect stats at the moment, but we sometimes grep through logs to have a general sense of whether something is used.  I created a new redirect: https://civicrm.org/go/stack-exchange  - it's version-controled in the nginx config, not in Drupal. I'll also add a monitoring check on it. (because yes, in the past, we've often had redirects that disappeared, but we've also hardcoded links to services/places that disappeared, so I think a monitored redirect is best)

File: templates/CRM/common/footer.tpl (hunk 1 above)
Comment (larssandergreen) [reply templates/CRM/common/footer.tpl:23]: Oops, done.

**Comment by larssandergreen** [comment]
Thanks @mlutfy
I changed the SE link.
I think for the System Status, probably more prominent makes sense for when it is in an error state, so in front, but I'm happy to change it too.

**Comment by yashodha** [comment]
@mlutfy do we have consensus to merge this PR or more needs to be done here?

**Comment by mlutfy** [comment]
I think it's good - but since it's a visual change (and changes to translation strings), would prefer to have a bit more feedback.

@JoeMurray ?

@elisseck ?

@joshgowans ?

(looking for deal-breakers, hoping to avoid bike-shed)

**Comment by joshgowans** [comment]
Sorry I've not followed the discussion, so my feedback probably should be discounted. Personally, I:

- would call it "Community Chat"
- prefer having the "Release notes" in-app somehow and don't like how we link out to a blog post, but that's a different issue
- think it's odd to have "About CiviCRM" under "Support". 

In general, I'm a fan of fewer links in general, so I think maybe a few aren't really necessary and could be removed. But, again, wasn't apart of the discussion so you can likely safely ignore the feedback.

**Comment by mlutfy** [comment]
@joshgowans Part of the PR review process is getting the feedback from people who might not be following the queue closely, and I think you're particularly qualified to comment here :)

I have to admit I had not really looked at the menu, and it's something admins/clients often mention as a source of confusion. Agree on keeping it short (it's a recurring complaint about CiviCRM menus).

Personally I would recommend:

Removing these items:

- More documentation (having "User Guide" is nice, and links to various places; the dev submenu has a link to the dev-guide)
- Release Notes (it will be confusing, they may not be on the latest version)
- Upcoming training (we don't have them regularly enough, it will be underwhelming)

Instead of "Ask a question" and "Chat", we should link to "Community help" and link here: https://civicrm.org/help

I would keep "About CiviCRM" - because people might be using a SaaS or inherit a CivICRM installation as part of their job.

Also keep "Report a bug" because it's a different form of engagement than "Community help", even if it's the same.

or more visually:

image

(I know that the items are not all verbs, but I'm trying to keep it short)

**Comment by elisseck** [comment]
@mlutfy oh wow yeah I was just a part of the conversation about the footer... I don't know that I have ever opened this menu XD. Your suggestions for removal sound reasonable and good to me.

I think the footer looks great without the extra clutter and the system status working... big thank you to @larssandergreen for making it happen :)

**Comment by larssandergreen** [comment]
@mlutfy I think we need to decide if the Support menu should have the support links or the support links should live on the website and the Support menu should just link to that page. A mix of both doesn't really make sense. It is confusing to have half the support links in the menu and then the other half on a page on the website, with some duplication. Personally, I think having them all in the Support menu is best — why make people click through to the website to see the options? Making SE and Mattermost more prominent seems like a good thing.

Happy to remove More documentation, Upcoming training and Release notes and to use 'Expert help' wording. I do think having Community chat per @joshgowans  and Ask a question makes sense. These should be at least as prominent as Report a bug, since we want people to use those before they report a bug.

Does anyone have any thoughts on updating existing installs? If not, I'll add to this PR as proposed above, adjusted for whatever the final list is.

**Comment by mlutfy** [comment]
hi @larssandergreen

About the chat:

- It's a good resource, but I don't like promoting it for general support because it does not help others, only specific cases, no archives. There are high odds no one will reply, or at least, not immediately.
- It's also why I prefer having menus for "Get expert help" vs "Get community help". When users click on that, the intent is clear. We can improve the website. It tends to be aimed too much at developers, not users/admins.

I know the website does not really do it at the moment, but I think there has to be somewhat clear expectations about Mattermost and about Stack Exchange, that they are managed by the community (for French I did this: https://civicrm.org/fr/aide)

The same applies to Stack Exchange: a lot of people don't know that it's community-managed, and that asking a question there for newbies is rather difficult/hostile (I don't think we realized that initially when we adopted SE).

And of course, a primary motivation is keeping it short. Maybe in different circumstances it would be different, but here I feel like it's not essential.

(sorry to be a pain) :-)

**Comment by larssandergreen** [comment]
What about just doing: User Guide, Get Help (which links to a page with all the options, including bug reporting) and About CiviCRM? Just keeping it simple and moving most of the details to the website then. I just think we shouldn't have a bunch of different places with a bunch of different selections of options, which makes it hard for someone to see all of them at once.

So either put them all on the website and have minimal options in the menu or put them all in the menu.

**Comment by mlutfy** [comment]
@larssandergreen Works for me :)

**Comment by eileenmcnaughton** [comment]
Just trying to see what this needs to get it over the line.

@mlutfy you have put quite a bit of effort into looking at this PR - I just skimmed it & seems like it is close - but perhaps we need a landing page on our docs or website for Get community help to point to?

**Comment by larssandergreen** [comment]
Forgot about this one for a bit, but I've now made the changes agreed above and have added database upgrades to remove the old support menu items and add the new ones.

@mlutfy I think [Get expert help](https://civicrm.org/experts) should be added to the [Find Support page](https://civicrm.org/help) on the website, so that all the help options are on one page. I think the "Self-service support resources" sub-heading there can be removed entirely, since it's all support options, not just self-service.

Also just want to verify that `https://docs.civicrm.org/user/?src=iam` is an acceptable URL for tracking purposes - or do we need to redirect that through `https://civicrm.org/`?

**Comment by larssandergreen** [comment]
For whatever reason the rebase isn't working for me today, so I've gone ahead and created a new PR (#26167), incorporating all feedback above.

**Pull request closed without merging**